### PR TITLE
 feat: add SageMaker model package promote modules

### PR DESCRIPTION
**Pull request by lccasagrande**
## Describe your changes

- Added `sagemaker-model-event-bus` module to setup event bridge buses with cross-account permissions.
- Added `sagemaker-model-package-group` module to create SageMaker Model Package Groups with a resource policy granting cross-account read-only access and event bridge rules to send SageMaker Model Package change events to another event bus.
- Added `sagemaker-model-package-promote-pipeline` module to create a CICD pipeline to automatically promote model packages across in a multi-account setup.

## Issue ticket number and link

N/A

## Checklist before requesting a review

- [X] I updated `CHANGELOG.MD` with a description of my changes
- [X] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [X] If the change was to a module, I have added thorough tests
- [X] If the change was to a module, I have added/updated the module's README.md
- [X] If a module was added, I added a reference to the module to the repository's README.md
- [X] I verified that my code deploys successfully using `seedfarmer apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: modules/sagemaker/sagemaker-model-event-bus/sagemaker_model_event_bus/settings.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    These parameters are required for the module stack.
+    """
+
+    model_config = SettingsConfigDict(env_prefix="SEEDFARMER_PARAMETER_")
+
+    event_bus_name: str
+
+    source_accounts: Optional[List[str]] = Field(default=None)
+    tags: Optional[Dict[str, str]] = Field(default=None)
```
Comment (LeonLuttenberger) [inline modules/sagemaker/sagemaker-model-event-bus/sagemaker_model_event_bus/settings.py:32]: This is really cool!

File: modules/sagemaker/sagemaker-model-package-group/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+### Input Paramenters
+
+#### Required
+
+- `model_package_group_name`: SageMaker Package Group Name to setup event rules.
+
+#### Optional
+
+- `retain_on_delete`: Wether or not to retain resources on delete. Defaults True.
```
Comment (LeonLuttenberger) [inline modules/sagemaker/sagemaker-model-package-group/README.md:21]: minor typo

File: modules/sagemaker/sagemaker-model-package-group/sagemaker_model_package_group/stack.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+        Returns
+        -------
+            A resource policy with cross-account read-only permissions.
+        """
+        if not self.target_account_ids:
+            return None
+
+        sagemaker_arn = f"arn:aws:sagemaker:{self.region}:{self.account}"
+        target_accounts = [f"arn:aws:iam::{a}:root" for a in self.target_account_ids]
```
Comment (LeonLuttenberger) [inline modules/sagemaker/sagemaker-model-package-group/sagemaker_model_package_group/stack.py:105]: We should use `self.partition` here to make the templates compatible with the China and US GOV partitions.

File: modules/sagemaker/sagemaker-model-package-promote-pipeline/sagemaker_model_package_promote_pipeline/stack.py [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+                    iam.PolicyStatement(
+                        sid="GrantSageMakerModelPkgReadOnlyPermissions",
+                        actions=[
+                            "sagemaker:DescribeModelPackageGroup",
+                            "sagemaker:DescribeModelPackage",
+                            "sagemaker:ListModelPackages",
+                        ],
+                        effect=iam.Effect.ALLOW,
+                        resources=[
+                            f"arn:aws:sagemaker:{self.region}:{self.source_account}:model-package/*",
```
Comment (LeonLuttenberger) [inline modules/sagemaker/sagemaker-model-package-promote-pipeline/sagemaker_model_package_promote_pipeline/stack.py:460]: Please use `self.partition` (same as above)

File: modules/sagemaker/sagemaker-model-package-promote-pipeline/sagemaker_model_package_promote_pipeline/stack.py [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+                        resources=[
+                            f"arn:{self.partition}:kms:{self.region}:{self.source_account}:key/*",
+                            f"arn:{self.partition}:kms:{self.region}:{self.account}:key/*",
+                        ],
+                    ),
+                    iam.PolicyStatement(
+                        sid="GrantS3ReadOnlyPermissions",
+                        actions=["s3:ListBucket", "s3:GetObject*"],
+                        effect=iam.Effect.ALLOW,
+                        resources=["arn:aws:s3:::*"],
```
Comment (LeonLuttenberger) [inline modules/sagemaker/sagemaker-model-package-promote-pipeline/sagemaker_model_package_promote_pipeline/stack.py:446]: Please use `self.partition` (same as above)

**Comment by kukushking** [comment]
@lccasagrande could you please update README.MD module table with the new modules? Also is it possible to keep `sagemaker-model-package-group` in `sagemaker` and move the event bus/pipeline to another group or examples/modules/ since these are not directly related to sagemaker?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: a86fd3b2e58e731205e1fe3c5644da6b386b4e30
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=44a4d068-2fa2-42cc-9e3c-b39b854654c2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Looks great, thanks @lccasagrande !

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 76f94a0bf071ccdefc94a86e311fe7de5b8192c1
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=490ab723-eb99-437c-9200-24b556226aa9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — DISMISSED** [review]

**Review by LeonLuttenberger — DISMISSED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 1084d453a51919181c881e804ded159a8b173a61
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=76f50315-d5ea-4918-ae88-37ec47c9f66a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 8e65044890049771bc5885a1e2746b214f82fe4d
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=9e74e944-0ada-4a93-a0df-9a616f1105f5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]

**Review by LeonLuttenberger — APPROVED** [review]

**Comment by kukushking** [comment]
Thanks @lccasagrande !

**Pull request merged**